nicegui/elements/progress.py

```diff
@@ -1,48 +1,124 @@
+        size: Optional[Union[str, int]] = 20,
+        show_value: bool = True,
+        color: Optional[str] = "primary",
+        reverse: bool = False,
+        buffer: Optional[float] = None,
+        track_color=None,
+        text_color: Optional[str] = "white",
+        dark: bool = False,
+        rounded: bool = False,
+        animation_speed: int = 2100,
+        indeterminate: bool = False,
+        query: bool = False,
+        instant_feedback: bool = False,
+        stripe: bool = False,
+    ) -> None:
         """Linear Progress
 
         A linear progress bar wrapping Quasar's
         `QLinearProgress <https://quasar.dev/vue-components/linear-progress>`_ component.
+        It provides a way to display a progress bar with customizable options.
 
-        :param value: the initial value of the field (from 0.0 to 1.0)
-        :param size: the height of the progress bar (default: "20px" with value label and "4px" without)
-        :param show_value: whether to show a value label in the center (default: `True`)
-        :param color: color (either a Quasar, Tailwind, or CSS color or `None`, default: "primary")
+        :param value: The initial value of the progress bar (from 0.0 to 1.0).
+        :param min: The minimum value of the progress bar (default: 0.0).
+        :param max: The maximum value of the progress bar (default: 1.0).
+        :param size: The height of the progress bar (default: None).
+        :param show_value: Whether to show a value label in the center (default: True).
+        :param color: The color of the progress bar (either a Quasar, Tailwind, or CSS color or `None`, default: "primary").
+        :param reverse: Whether to reverse the direction of the progress bar (default: False).
+        :param buffer: The buffer value of the progress bar (default: None).
+        :param track_color: The color of the progress bar track (either a Quasar, Tailwind, or CSS color or `None`, default: None).
+        :param dark: Whether to use dark mode for the progress bar (default: False).
+        :param rounded: Whether to use rounded corners for the progress bar (default: False).
+        :param animation_speed: The animation speed of the progress bar (default: 2100).
+        :param indeterminate: Whether the progress bar is indeterminate (default: False).
+        :param query: Whether the progress bar is in query mode (default: False).
+        :param instant_feedback: Whether to provide instant feedback for the progress bar (default: False).
+        :param stripe: Whether to show a stripe pattern on the progress bar (default: False).
         """
-        super().__init__(tag='q-linear-progress', value=value, on_value_change=None, text_color=color)
-        self._props['size'] = size if size is not None else '20px' if show_value else '4px'
+        if max > 1.0:
+            value = float(value / max)
+
+        super().__init__(
```

Comment:
Please don't reformat lines that are unrelated to this PR.
There are more such format changes below.